client/me/purchases/manage-purchase/index.jsx

```diff
@@ -373,13 +378,42 @@ class ManagePurchase extends Component {
 		const isUpgradeableBackupProduct = JETPACK_BACKUP_T1_PRODUCTS.includes( purchase.productSlug );
 		const isUpgradeableSecurityPlan = JETPACK_SECURITY_T1_PLANS.includes( purchase.productSlug );
 
+		if ( isAkismetProduct( purchase ) ) {
+			return this.getAkismetUpgradeUrl();
+		}
+
 		if ( isUpgradeableBackupProduct || isUpgradeableSecurityPlan ) {
 			return `/plans/storage/${ siteSlug }`;
 		}
 
 		return `/plans/${ siteSlug }`;
 	}
 
+	// For the first Iteration of Calypso Akismet checkout we are only suggesting
+	// for immediate upgrades to the next plan. We will change this in the future
+	// with appropriate page.
+	getAkismetUpgradeUrl() {
+		const { purchase } = this.props;
+
+		const monthlyIndex = AKISTMET_PLUS_MONTHLY_PRODUCTS.indexOf( purchase.productSlug );
+		if ( monthlyIndex === AKISTMET_PLUS_MONTHLY_PRODUCTS.length - 1 ) {
+			return 'https://akismet.com/enterprise';
+		} else if ( monthlyIndex !== -1 ) {
+			const nextPlan = AKISTMET_PLUS_MONTHLY_PRODUCTS[ monthlyIndex + 1 ];
+			return `/checkout/akismet/${ nextPlan }`;
+		}
+
+		const yearlyIndex = AKISMET_PLUS_YEARLY_PRODUCTS.indexOf( purchase.productSlug );
+		if ( yearlyIndex === AKISMET_PLUS_YEARLY_PRODUCTS.length - 1 ) {
+			return 'https://akismet.com/enterprise';
+		} else if ( yearlyIndex !== -1 ) {
+			const nextPlan = AKISMET_PLUS_YEARLY_PRODUCTS[ yearlyIndex + 1 ];
```

Comment:
This approach works as long as we can assume the order of `AKISMET_PLUS_YEARLY_PRODUCTS ` matches our desired upgrade paths. Would there ever be an instance where that is not true?

An upgrade map would be much more verbose, but perhaps easier to parse? WDYT?

Reply:
Yes, I could define the Upgrade map. Initially I've done this way to use current logic as my assumption that we would want to create a proper Upsell page within month or two, and then the we don't need the mapping